lib/server.js

```diff
@@ -68,53 +47,87 @@ function createServer (options, httpHandler) {
       } else {
         /* Deal with listen ([port[, host[, backlog]]]) */
         options.port = argsLength >= 1 && firstArg ? firstArg : 0
-        // This will listen to what localhost is.
-        // It can be 127.0.0.1 or ::1, depending on the operating system.
-        // Fixes https://github.com/fastify/fastify/issues/1022.
-        options.host = argsLength >= 2 && args[1] ? args[1] : 'localhost'
+        options.host = host
         options.backlog = argsLength >= 3 ? args[2] : undefined
       }
 
       return options
     }
 
+    const bindedListen = _listen.bind(this)
     const listenOptions = normalizeListenArgs(Array.from(arguments))
+
+    let secondHost
+    let clonedOptions
+    // TODO: Remove once finished
+    process._rawDebug('Binding second host:', listenOptions.host === '127.0.0.1' || listenOptions.host === '::1')
+    if (listenOptions.host === '127.0.0.1' || listenOptions.host === '::1') {
+      const bindingHost = listenOptions.host === '127.0.0.1' ? '::1' : '127.0.0.1'
```

Comment:
What if ::1 is not available in the system? We need to verify that first.

Reply:
Sure, I'm aware that using [`os.networkInterfaces()`](https://nodejs.org/api/os.html#osnetworkinterfaces) can be a way to go, but it's deterministic in a way that it works exactly the same for both Windows/Linux/Mac? If so, just looking for `lo` + `::1` should do the work, or is there any other way?

Reply:
we would need to verify it across operating systems.

Reply:
Any suggestion about how to properly test it on Windows? I only have the rough idea of a VM

Reply:
Using `Promise.any` can be another option, as will silently ignore such issues when trying to set both instances. Not sure if it's a good idea though